libbeat/docs/security/securing-beats.asciidoc

```diff
@@ -1,33 +1,82 @@
+To send data to a secured cluster through the `elasticsearch` output,
+{beatname_uc} needs to authenticate as a user who can manage index templates,
+monitor the cluster, create indices, read and write to the indices
+it creates, and manage ingest pipelines. 
+endif::[]
+ifeval::["{beatname_lc}"!="filebeat"]
 To send data to a secured cluster through the `elasticsearch` output,
 {beatname_uc} needs to authenticate as a user who can manage index templates,
-monitor the cluster, create indices, and read, and write to the indices
-it creates. See <<beats-basic-auth>>.
+monitor the cluster, create indices, and read and write to the indices
+it creates. 
+endif::[]
 
-If encryption is enabled on the cluster, you also need to enable HTTPS in the
-{beatname_uc} configuration. See <<beats-tls>>.
+. <<beats-user-access>>.
++
+To search the indexed {beatname_uc} data and visualize it in {kib}, users need
+access to the indices {beatname_uc} creates.
 
-In addition to configuring authentication credentials for the {beatname_uc}
-itself, you need to grant authorized users permission to access the indices it
-creates. See <<beats-user-access>>.
+. <<beats-tls>>.
++
+If encryption is enabled on the cluster, you need to enable HTTPS in the
+{beatname_uc} configuration.
 
 ifeval::["{beatname_lc}"!="apm-server"]
-If you plan to monitor {beatname_uc} in Kibana, you must also 
-<<beats-system-user,configure the `beats_system` built-in user>>.
+. <<beats-system-user>>.
++
+{beatname_uc} uses the `beats_system` user to send monitoring data to {es}. If
+you plan to monitor {beatname_uc} in {kib} and have not yet set up the
+password, set it up now.
 endif::[]
 
 For more information about {security}, see
-{xpack-ref}/xpack-security.html[Securing {es} and {kib}].
+{xpack-ref}/xpack-security.html[Securing the {stack}].
+
+[float]
+=== {beatname_uc} features that require authorization
+
+After securing {beatname_uc}, make sure your users have the roles (or associated
+privileges) required to use these {beatname_uc} features:
+
+[options="header"]
+|=======
+|Feature | Role footnote:[You must create the ++{beat_default_index_prefix}_writer++
+and +{beat_default_index_prefix}_reader+ roles (see <<beats-basic-auth>> and
+<<beats-user-access>>.) The `machine_learning_admin` and `kibana_user` roles
+are {xpack-ref}/built-in-roles.html[built-in].]
+|Send data to a secured cluster   | ++{beat_default_index_prefix}_writer++
+|Run Filebeat modules | ++{beat_default_index_prefix}_writer++
+|Load index templates | ++{beat_default_index_prefix}_writer++ and `kibana_user`
+|Load {beatname_uc} dashboards into {kib} | ++{beat_default_index_prefix}_writer++ and `kibana_user`
+|Load machine learning jobs | `machine_learning_admin`
+|Read indices created by {beatname_uc} | ++{beat_default_index_prefix}_reader++ 
+|View {beatname_uc} dashboards in {kib} | `kibana_user`
+|=======
+
+// REVIEWERS: Should I mention machine learning jobs? TBH, I'm not sure which
```

Comment:
I would keep ML. At the moment I think only Filebeat has them, but hopefully more Beats get ML configs soon.